de.peeeq.wurstscript/src/main/java/de/peeeq/wurstio/languageserver/requests/Colors.java

```diff
@@ -100,18 +101,24 @@ public ColorPresentationRequest(ColorPresentationParams params) {
 
         @Override
         public List<ColorPresentation> execute(ModelManager modelManager) {
-            CompilationUnit cu = modelManager.getCompilationUnit(WFile.create(textDocument));
-            if (cu == null) {
+            Optional<CompilationUnit> cu = Optional.ofNullable(
+                modelManager.getCompilationUnit(WFile.create(textDocument)));
+            if (cu.isEmpty()) {
                 return Collections.emptyList();
             }
-            Element elem = Utils.getAstElementAtPos(cu, range.getStart().getLine() + 1, range.getStart().getCharacter() + 2, false);
+            Optional<Element> elemAtPos = Utils.getAstElementAtPos(cu.get(), range.getStart().getLine() + 1,
+                range.getStart().getCharacter() + 2, false);
+            if (elemAtPos.isEmpty()) {
+                return Collections.emptyList();
+            }
+            Element elem = elemAtPos.get();
 
             if (elem instanceof ExprIntVal
-                    && elem.getParent() != null) {
+                    && Optional.ofNullable(elem.getParent()).isPresent()) {
```

Comment:
@peq I see that Wscope.java is AST-generated. Is there any way to make it return Optional for getParent?